### DISCUSSION TITLE
Configuration node

### DISCUSSION THREAD

**Discussion by rapzak** [leon-thomm/Ryven#112] [Q&A]
Hi,
I am thinking and trying out to make an electronic lab framework based on this flow, to commnicate with many different measurement devices - for fast automated tests...
One example could be i need to talk to many different instruments on a GPIB bus.
Then i like in flow to have a config block to setup the bus communication, make a running class that handles this communication, like the configuration node in node-red. 
This class need to be called from a lot of the other nodes, to execute commands or read back data.
Is there a good practise where to generate this "global" class (it has to happen automatic when the config block is added)

Thank you

**Comment by leon-thomm** [comment]
Hey, I'm not sure what that configuration node is supposed to do but it sounds like you just want something static. You can simply put that into your nodes package, e.g.

`gpib/nodes.py`
```python
from ryven.NENV import *

class Interface:
    # your nodes' interface to the busses
    ...

# static module-level interface object which all nodes can access
interface = Interface()

class SendData_Node(Node):
    def __init__(self, params):
        super().__init__(params)

        self.ports = interface.scan()
        ...

...

export_nodes(SendData_Node, ...)
```
something like that?
